src/core/batch.rkt

```diff
@@ -1,15 +1,19 @@
 #lang racket
 
 (require "../utils/timeline.rkt"
-         "../syntax/syntax.rkt")
+         "../syntax/syntax.rkt"
+         "../syntax/types.rkt")
 
-(provide progs->batch
-         batch->progs
+(provide progs->batch ; (Listof Expr) -> Batch
+         batch->progs ; Batch -> (Listof Expr)
          (struct-out batch)
-         batch-length
-         batch-ref
-         deref
-         batch-replace)
+         (struct-out batchref) ; temporarily for patch.rkt
+         batch-length ; Batch -> Integer
+         batch-ref ; Batch -> Index -> Expr
+         deref ; Batchref -> Expr
+         batch-replace ; Batch -> Lambda -> Batch
```

Comment:
Fine to write `Lambda` but specifically it is `(Expr -> Expr)` or even `(Expr<Batchref> -> Expr<Batchref>)` right?